roslibrust_codegen/src/gen.rs

```diff
@@ -145,20 +147,48 @@ fn generate_field_definition(
         )?;
         if field.field_type.array_info.is_some() {
             // For vectors use smart_defaults "dynamic" style
-            Ok(quote! {
+            quote! {
                 #[default(_code = #default_val)]
-                pub #field_name: #rust_field_type,
-            })
+            }
         } else {
             // For non vectors use smart_default's constant style
-            Ok(quote! {
+            quote! {
               #[default(#default_val)]
-              pub #field_name: #rust_field_type,
-            })
+            }
         }
     } else {
-        Ok(quote! { pub #field_name: #rust_field_type, })
-    }
+        // Okay this is messy, so default isn't defined for fixed sized arrays > 32 in length
+        // so we have to manually provide a default if one isn't provided for arrays that large
+        if let Some(Some(fixed_array_length)) = field.field_type.array_info {
+            if fixed_array_length > 32 {
+                // Doing some evil indirection here with the _code directive and Deafult::default()
+                // to generate the default value for a single member of the array type, and then
+                // broadcasting that with an array constant. I can't believe this works...
+                let default_str = format!("[Default::default(); {fixed_array_length}]");
+                quote! { #[default(_code = #default_str)]}
+            } else {
+                quote! {}
+            }
+        } else {
+            quote! {}
+        }
+    };
+    let serde_line = if let Some(Some(fixed_array_length)) = field.field_type.array_info {
```

Comment:
Nitpick: This is a little unclean looking, perhaps:
```rust
let serde_line = match field.field_type.array_info {
    Some(Some(fixed_array_len)) if fixed_array_len > MAX_FIXED_ARRAY_LEN => quote! { #[serde(with = "::serde_big_array::BigArray")] },
    _ => quote! {},
};
```

Reply:
Love that much cleaner.